c_glib/arrow-glib/basic-array.cpp

```diff
@@ -1039,6 +1048,32 @@ garrow_int8_array_get_values(GArrowInt8Array *array,
   return garrow_array_get_values_raw<arrow::Int8Type>(arrow_array, length);
 }
 
+/**
+ * garrow_int8_array_sum:
+ * @array: A #GArrowInt8Array.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: Returns: The value of the computed sum on success,
+ *   0 on error.
```

Comment:
"0 on error" is misleading. 0 is one of a valid value. Users should not use 0 as an error indicator.

Reply:
I've fixed documents.

Reply:
Note that it may also return NULL.

Reply:
Thanks.
We need to NULL check, then.

Reply:
I've added.
We use zero for the case because we can't implement `option<T>` in C for primitive types.